src/Rules/Exceptions/DefaultExceptionTypeResolver.php

```diff
@@ -27,6 +28,16 @@ public function __construct(
 		private array $checkedExceptionClasses,
 	)
 	{
+		foreach ($this->checkedExceptionClasses as $checkedExceptionClass) {
+			if (!$this->reflectionProvider->hasClass($checkedExceptionClass)) {
+				throw new ShouldNotHappenException('Class ' . $checkedExceptionClass . ' used in \'checkedExceptionClasses\' does not exist.');
+			}
+		}
+		foreach ($this->uncheckedExceptionClasses as $uncheckedExceptionClass) {
+			if (!$this->reflectionProvider->hasClass($uncheckedExceptionClass)) {
+				throw new ShouldNotHappenException('Class ' . $uncheckedExceptionClass . ' used in \'uncheckedExceptionClasses\' does not exist.');
+			}
+		}
```

Comment:
Do you want that behind featureFlag / bleedingEdge? I think this will be pretty rare.